cypress-io/cypress
# feat: add RecordRunModal component

- Closes #23764

### User facing changelog


n/a

### Additional details


This PR adds the `RecordRunModal` UI component and refactors some logic from `RecordBanner` to be reusable between these two components. The work to integrate this component into the UI will be done in #23766

### Steps to test


The component isn't actually used anywhere in the UI yet, so take a look at the test

1. Run `RecordRunModal.cy.tsx` in `frontend-shared`
2. Look at the component and ensure that it meets all of the criteria laid out in the issue
3. Verify that `RecordBanner` still works as before

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
